2k48/2k48/src/BoardLogic.js

```diff
@@ -0,0 +1,241 @@
+                (current === 0) ? row.unshift(current) : row.push(current);
+            }
+            board.push(row);
+        }
+
+        for (let i = 0; i < board.length; i++) {
+            for (let j = board[i].length - 1; j >= 0; j--) {
+                if (board[i][j] > 0 && board[i][j] === board[i][j - 1]) {
+                    board[i][j] = board[i][j] * 2;
+                    board[i][j - 1] = 0;
+                    score += board[i][j];
+                } else if (board[i][j] === 0 && board[i][j - 1] > 0) {
+                    board[i][j] = board[i][j - 1];
+                    board[i][j - 1] = 0;
+                }
+            }
+        }
+
+        return { board, score };
+    }
+
+    moveDown(inputBoard) {
+        let rotatedRight = this.rotateRight(inputBoard);
+        let { board, score } = this.moveLeft(rotatedRight);
+        board = this.rotateLeft(board);
+
+        return { board, score };
+    }
+
+    moveLeft(inputBoard) {
+        let board = [];
+        let score = 0;
+
+        for (let r = 0; r < inputBoard.length; r++) {
+            let row = [];
+            for (let c = inputBoard[r].length - 1; c >= 0; c--) {
+                let current = inputBoard[r][c];
+                (current === 0) ? row.push(current) : row.unshift(current);
+            }
+            board.push(row);
+        }
+
+        for (let r = 0; r < board.length; r++) {
+            for (let c = 0; c < board.length; c++) {
+                if (board[r][c] > 0 && board[r][c] === board[r][c + 1]) {
+                    board[r][c] = board[r][c] * 2;
+                    board[r][c + 1] = 0;
+                    score += board[r][c];
+                } else if (board[r][c] === 0 && board[r][c + 1] > 0) {
+                    board[r][c] = board[r][c + 1];
+                    board[r][c + 1] = 0;
+                }
+            }
+        }
+
+        return { board, score };
+    }
+
+    rotateRight(matrix) {
+        let result = [];
+
+        for (let c = 0; c < matrix.length; c++) {
+            let row = [];
+            for (let r = matrix.length - 1; r >= 0; r--) {
+                row.push(matrix[r][c]);
+            }
+            result.push(row);
+        }
+
+        return result;
+    }
+
+    rotateLeft(matrix) {
+        let result = [];
+
+        for (let c = matrix.length - 1; c >= 0; c--) {
+            let row = [];
+            for (let r = matrix.length - 1; r >= 0; r--) {
+                row.unshift(matrix[r][c]);
+            }
+            result.push(row);
+        }
+
+        return result;
+    }
+
+    checkForGameOver(board) {
+        let moves = [
+            this.boardMoved(board, this.moveUp(board).board),
+            this.boardMoved(board, this.moveRight(board).board),
+            this.boardMoved(board, this.moveDown(board).board),
+            this.boardMoved(board, this.moveLeft(board).board)
+        ];
+
+        return !moves.includes(true);
```

Comment:
use moves.every(move => move === false). I think it reveals the intention better